File: openmp/testsuite/fortran/omp_in_parallel.f
<ompts:test>
<ompts:testdescription>Test which checks that omp_in_parallel returns false when called from a serial region and true when called within a parallel region.</ompts:testdescription>
<ompts:ompversion>2.0</ompts:ompversion>
<ompts:directive>omp_in_parallel</ompts:directive>
<ompts:testcode>
      INTEGER FUNCTION <ompts:testcode:functionname>omp_in_parallel</ompts:testcode:functionname>()
!   checks that false is returned when called from serial region
!   and true is returned when called within parallel region
        LOGICAL omp_in_parallel
		<ompts:orphan:vars>
!        LOGICAL omp_in_parallel
        LOGICAL serial, parallel
        COMMON /orphvars/ serial, parallel
		</ompts:orphan:vars>
        serial=.TRUE.
        parallel=.FALSE.

		<ompts:orphan>
		<ompts:check>
        serial=omp_in_parallel()
		</ompts:check>
		</ompts:orphan>

!$omp parallel
!$omp single
		<ompts:orphan>
		<ompts:check>
        parallel=omp_in_parallel();
		</ompts:check>
		</ompts:orphan>
!$omp end single
!$omp end parallel

        IF ( (.NOT. serial) .AND. (parallel) ) THEN
          <testfunctionname></testfunctionname>=1
        ELSE
          <testfunctionname></testfunctionname>=0
        END IF
      END FUNCTION
</ompts:testcode>
</ompts:test>

File: openmp/testsuite/c/omp_in_parallel.c
<ompts:test>
<ompts:testdescription>Test which checks that omp_in_parallel returns false when called from a serial region and true when called within a parallel region.</ompts:testdescription>
<ompts:ompversion>2.0</ompts:ompversion>
<ompts:directive>omp_in_parallel</ompts:directive>
<ompts:testcode>
/*
 * Checks that false is returned when called from serial region
 * and true is returned when called within parallel region. 
 */
#include <stdio.h>
#include "omp_testsuite.h"

int <ompts:testcode:functionname>omp_in_parallel</ompts:testcode:functionname>(FILE * logFile){
    <ompts:orphan:vars>
	int serial;
	int isparallel;
    </ompts:orphan:vars>

    serial = 1;
    isparallel = 0;

    <ompts:check>
	<ompts:orphan>
	    serial = omp_in_parallel ();
	</ompts:orphan>

#pragma omp parallel
    {
#pragma omp single
	{
	    <ompts:orphan>
		isparallel = omp_in_parallel ();
	    </ompts:orphan>
	}
    }
    </ompts:check>

    <ompts:crosscheck>
#pragma omp parallel
	{
#pragma omp single
	    {

	    }
	}
    </ompts:crosscheck>

	return (!(serial) && isparallel);
}
</ompts:testcode>
</ompts:test>
